Remove flacky test and replace with Mocking test

davidchencsl opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #80098
* #80094
* #79993
* __->__ #80159
* #79974

This PR removes the flaky test that was introduce in the previous commit, and replace it with a mocking test. The actual test of queue depth is added in the next commit.

davidchencsl committed:
Remove flacky test and replace with Mocking test

[ghstack-poisoned]

davidchencsl committed:
Update on "Remove flacky test and replace with Mocking test"




[ghstack-poisoned]

davidchencsl committed:
Update on "Remove flacky test and replace with Mocking test"




[ghstack-poisoned]

robieta commented on test/test_profiler.py:
```diff
@@ -1192,5 +1140,109 @@ def find_ones(nodes):
+class TestExperimentalUtils(TestCase):
+
+    @staticmethod
+    def generate_mock_profile():
```
This is an absolutely beautiful method. You should include a unit test that utilizes it.

robieta replied:
I see you use it in later PRs to test the new functionality, but you should also use it to validate the existing functionality.

davidchencsl replied:
Okay understood

davidchencsl replied:
I added a temporary test (with a list of expected digits) because I'm missing some information in qd list to print GPU/CPU/cudaLaunch in this commit. This should be enough to verify correctness tho.

davidchencsl committed:
Update on "Remove flacky test and replace with Mocking test"


This PR removes the flaky test that was introduce in the previous commit, and replace it with a mocking test. The actual test of queue depth is added in the next commit.

[ghstack-poisoned]

robieta reviewed (approved):
Thanks!

davidchencsl commented:
@pytorchmergebot please merge

davidchencsl commented:
@pytorchmergebot merge
